src/acceptance-test/java/org/zalando/nakadi/webservice/hila/SubscriptionLoadAT.java

```diff
@@ -38,51 +39,46 @@
-        clients.forEach(TestStreamingClient::start);
-        timeLogger.mark("Create " + clients.size() + " clients");
-
-        // stream all the data.
-        final Map<String, String> receivedEvents = streamAllData(
-                clients, MESSAGES_PER_EVENT_TYPE * EVENT_TYPE_COUNT);
-        timeLogger.mark("Receive and commit " + receivedEvents.size() + " events");
-
-        // Close all connections
-        clients.parallelStream().forEach(TestStreamingClient::close);
-        timeLogger.mark("Closing connections");
-
-        // Check total events count
-        Assert.assertEquals(EVENT_TYPE_COUNT * MESSAGES_PER_EVENT_TYPE, receivedEvents.size());
-        // Check that every consumer received at least something
-        clients.stream().map(TestStreamingClient::getSessionId).forEach(
-                sessionId -> Assert.assertTrue(
-                        receivedEvents.values().stream().filter(v -> v.equals(sessionId)).count() > 0));
+        TimeLogger.startMeasure("testSubscriptionLoad", "createEventTypes");
+        try {
+            final List<EventType> eventTypeList = createEventTypes(EVENT_TYPE_COUNT, PARTITION_PER_EVENT_TYPE);
+
+            // publish events to each event type
+            TimeLogger.addMeasure("Publishing " + (MESSAGES_PER_EVENT_TYPE * EVENT_TYPE_COUNT) + " messages");
+            eventTypeList.parallelStream()
+                    .forEach(et -> sendEvents(et, MESSAGES_PER_EVENT_TYPE, PARTITION_PER_EVENT_TYPE));
+
+            // create subscription
+            TimeLogger.addMeasure("Create subscription");
+            final Subscription subscription = createSubscription(eventTypeList);
+
+            TimeLogger.addMeasure("Create clients");
+            final List<TestStreamingClient> clients = createSteamingClients(
+                    subscription, Math.min(EVENT_TYPE_COUNT + 5, EVENT_TYPE_COUNT * PARTITION_PER_EVENT_TYPE));
+            clients.forEach(TestStreamingClient::start);
+
+            // stream all the data.
+            TimeLogger.addMeasure("Receive and commit events");
+            final Map<String, String> receivedEvents = streamAllData(
+                    clients, MESSAGES_PER_EVENT_TYPE * EVENT_TYPE_COUNT);
+
+            // Close all connections
+            TimeLogger.addMeasure("Closing connections");
+            clients.parallelStream().forEach(TestStreamingClient::close);
+
+            TimeLogger.addMeasure("checking");
+            // Check total events count
+            Assert.assertEquals(EVENT_TYPE_COUNT * MESSAGES_PER_EVENT_TYPE, receivedEvents.size());
+            // Check that every consumer received at least something
+            clients.stream().map(TestStreamingClient::getSessionId).forEach(
+                    sessionId -> Assert.assertTrue(
+                            receivedEvents.values().stream().filter(v -> v.equals(sessionId)).count() > 0));
+        } finally {
+            System.out.println(TimeLogger.finishMeasure());
```

Comment:
Cause this test includes some kind of load testing. Initially it was printing lines. Now it is doing the same using TimeLogger infrastructure.